### PR TITLE
[WIP] Upload perf artifacts to Azure Blob Storage using AzCopy

### PR DESCRIPTION
**Pull request by adamsitnik**

### CODE REVIEW

**Comment by adamsitnik** [comment]
@dotnet-bot test ci please

**Comment by adamsitnik** [comment]
@dotnet-bot test ci please

**Comment by adamsitnik** [comment]
@dotnet-bot test ci please

**Comment by adamsitnik** [comment]
@dotnet-bot test ci please

File: tests/scripts/azcopy/Install-AzCopy.ps1 [hunk 1]
```diff
@@ -0,0 +1,32 @@
+Param(
+  [string] $toolsPath = "$PSScriptRoot\..\..\..\Tools\azcopy"
+)
+
```
Comment (jorive) [inline tests/scripts/azcopy/Install-AzCopy.ps1]: I would add there lines to the powershell scripts:
```powershell
Set-StrictMode -Version Latest
$ErrorActionPreference = "Stop"
```

File: tests/scripts/azcopy/Install-AzCopy.ps1 [hunk 2]
```diff
@@ -0,0 +1,32 @@
+Param(
```
Comment (jorive) [inline tests/scripts/azcopy/Install-AzCopy.ps1]: I would add these lines at the top of the powershell scripts:
```powershell
#Requires -Version 4
[CmdletBinding()]
Param(
```

**Comment by jorive** [comment]
Given that you are trying to install a tool, should we expect the tool to be on the machine already? Add it to the machine setup script instead?
@DrewScoggins thoughts?
AzCopy is also available through [https://chocolatey.org/packages/azcopy](https://chocolatey.org/packages/azcopy) (not sure if it is the right version)

File: perf.groovy [hunk 3]
```diff
@@ -100,18 +101,32 @@ def static getOSGroup(def os) {
                                     batchFile("py tests\\scripts\\run-xunit-perf.py ${runXUnitPerfCommonArgs} -testBinLoc bin\\tests\\${os}.${architecture}.${configuration}\\Jit\\Performance\\CodeQuality -collectionFlags default+BranchMispredictions+CacheMisses+InstructionRetired+gcapi")
                                 }
                             }
+
```
Comment (jorive) [inline perf.groovy]: >	 [](start = 0, length = 1)

nit: Replace Tab with spaces.

**Comment by jorive** [comment]
The equivalent bash files are missing.

**Comment by adamsitnik** [comment]
@dotnet-bot test ci please

**Comment by adamsitnik** [comment]
@dotnet-bot test ci please

**Comment by adamsitnik** [comment]
@dotnet-bot test ci please

**Comment by adamsitnik** [comment]
@dotnet-bot test ci please

File: tests/scripts/azcopy/Install-AzCopy.ps1 [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+	Write-Host "Copying AzCopy to $toolsPath"
+	New-Item -ItemType directory -Path $toolsPath | Out-Null
+	Copy-Item "$unzipped\Microsoft SDKs\Azure\AzCopy\*" $toolsPath -Force
+
+	Remove-Item $temp -Recurse -Force
+}
+
+# Display version of AzCopy.exe downloaded
+Get-ChildItem $azcopy |% VersionInfo | Select ProductVersion,FileVersion
```
Comment (jorive) [inline tests/scripts/azcopy/Install-AzCopy.ps1]: >Select [](start = 39, length = 6)

Please replace with `Select-Object`

**Comment by jorive** [comment]
@dotnet-bot test ci please

**Pull request closed without merging**